ludwig/encoders/image_torchvision_encoders.py

```diff
@@ -0,0 +1,740 @@
+            )
+        else:
+            weights_specification = None
+            if saved_weights_in_checkpoint:
+                logger.info(
+                    f"Instantiating torchvision image encoder: '{self.torchvision_model_type}' "
+                    "with weights saved in the checkpoint."
+                )
+            else:
+                logger.info(
+                    f"Instantiating torchvision image encoder: '{self.torchvision_model_type}' "
+                    "with no pretrained weights."
+                )
+
+        # get torchvision transforms object
+        transforms_obj = torchvision_model_registry[self.torchvision_model_type][
+            self.model_variant
+        ].model_weights.DEFAULT.transforms()
+        self.num_channels = len(transforms_obj.mean)
+        self.crop_size = transforms_obj.crop_size
+
+        logger.debug(f"  {self.torchvision_model_type}")
+        # create pretrained model with pretrained weights or None for untrained model
+        self.model = self.create_model(weights=weights_specification, **kwargs)
+
+        # remove final classification layer
+        self._remove_softmax_layer()
+
+        # freeze parameters if requested
+        for p in self.model.parameters():
+            p.requires_grad_(trainable)
+
+    def forward(self, inputs: torch.Tensor) -> Dict[str, torch.Tensor]:
+        return {"encoder_output": self.model(inputs)}
+
+    @abstractmethod
+    def _remove_softmax_layer(self):
+        """Model specific method that allows the final softmax layer to be implemented in the Ludwig Decoder
+        component.  The model specific implementation should change the final softmax layer in the torchvision
+        model architecture to torch.nn.Identity().  This allows the output tensor from the preceding layer to be
+        passed to the Ludwig Combiner and then to the Decoder.
+
+        Returns: None
+        """
+        raise NotImplementedError()
+
+    @property
+    def output_shape(self) -> torch.Size:
+        # create synthetic image and run through forward method
+        inputs = torch.randn([1, *self.input_shape])
+        output = self.model(inputs)
+        return torch.Size(output.shape[1:])
+
+    @property
+    def input_shape(self) -> torch.Size:
+        # expected shape after all pre-processing
+        # len(transforms_obj.mean) determines the number of channels
+        # transforms_obj.crop_size determines the height and width of image
+        # [num_channels, height, width]
+        return torch.Size([self.num_channels, *(2 * self.crop_size)])
+
+    @classmethod
+    def is_pretrained(cls, encoder_params: Dict[str, Any]) -> bool:
+        return encoder_params.get("use_pretrained", True)
+
+
+@DeveloperAPI
+@register_torchvision_model_variants(
+    [
+        TVModelVariant(variant_id="base", create_model_function=tvm.alexnet, model_weights=tvm.AlexNet_Weights),
+    ]
+)
+@register_encoder("alexnet_torch", IMAGE)
```

Comment:
Can we remove the `_torch` suffix from these? Most users should not need to know that these are from torchvision specifically. Rather it's an implementation detail. This also allows us to transfer the `resenet` and `vit` implementations that were removed to the new classes without breaking previous configs.

Reply:
The `_torch` suffix was added after the community meeting in October.   This was @w4nderlust  suggestion and is intended to lay the foundation for eventually including other pre-trained models, like Hugging Face, which would get the a `_hf` suffix.